src/TinyCiv.Client/Code/MVVM/MainViewModel.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TinyCiv.Client.Code.Core;
+using TinyCiv.Shared.Game;
+
+namespace TinyCiv.Client.Code.MVVM
+{
+    public class MainViewModel : ObservableObject
+    {
+        private TeamColor _color;
+        public TeamColor PlayerColor
+        { 
+            get
+            {
+                return _color;
+            }
+            set
+            {
+                _color = value;
+                OnPropertyChanged();
+            }
+        }
```

Comment:
@Randwarf just a suggestion

You could create something like (might have syntax errors):
```
public class ObservableObject<T> : where T : IEquatable
{
     private T _value;
     private Action someAction;


     T Value 
     {
        get { return _value;}
        set {
             if (!_value.Equal(value))
                    someAction()  
             }

              _value = value;
      }

      public void OnChange(Action someAction)
     {
           someAction();
     }
 }
```
and then use like:
```
public ObservableObject<Position> Position;
Position.OnChange(() => do some stuff)
```

just something to consider

Reply:
can also pass in new/old value into that ```Action``` etc